devices/yale.js

```diff
@@ -20,6 +20,41 @@ const lockExtend = (meta, lockStateOptions=null, binds=['closuresDoorLock', 'gen
     };
 };
 
+const fzLocal = {
+    c4_lock_operation_event: {
+        cluster: 'genAlarms',
+        type: ['commandAlarm'],
+        convert: async (model, msg, publish, options, meta) => {
+            const result = {};
+            if (msg.data.clusterid == 64512) {
+                // We need to read the lock state in case the alarm code is unknown
```

Comment:
I've added the try catch. One thing I'm wondering:

> // We need to read the lock state in case the alarm code is unknown

Can this read move to `if (!lookup[alarmcode]) {`? (given that we only need to read when unknown)

Reply:
Thanks!

I could put it there and add another lookup / condition statement to determine lock state - I was just replicating how the module communicates with it's native hub.

Let me know if you want me to change it :-)